network/crl.go

```diff
@@ -88,29 +88,58 @@ func (c DefaultCRLClient) Fetch(uri string) (*pkix.CertificateList, error) {
 			return nil, err
 		}
 
-		crl, err := x509.ParseCRL(content)
-		if err != nil {
-			return nil, err
-		}
-
-		return crl, nil
+		return content, nil
 	} else if strings.HasPrefix(uri, "file://") {
 		content, err := ioutil.ReadFile(uri[7:])
 		if err != nil {
 			return nil, err
 		}
 
-		crl, err := x509.ParseCRL(content)
-		if err != nil {
-			return nil, err
-		}
-
-		return crl, nil
+		return content, nil
 	}
 
 	return nil, errors.New(fmt.Sprintf("Cannot fetch CRL from '%s', unsupported protocol", uri))
 }
 
+type CRLCache interface {
+	Get(key string) ([]byte, error)
+	Put(key string, value []byte) error
+	Remove(key string) error
+}
+
+type DefaultCRLCache struct {
+	cache sync.Map
+}
+
+func (c *DefaultCRLCache) Get(key string) ([]byte, error) {
+	log.Debugf("CRL: cache: loading '%s'", key)
+	value, ok := c.cache.Load(key)
+	if ok {
+		value, ok := value.([]byte)
+		if !ok {
+			// should never happen since Put() only inserts []byte
+			return nil, errors.New("Unexpected value of invalid type")
```

Comment:
can I ask you to add unit-test for this cache with check for this execution branch with this error?)

Reply:
You mean artificially put some other type there and then `Get()` it?

This branch should be unreachable because as comment says we only use `Put()` to store things and `Put()` is limited to this particular type (`*CRLCacheItem`)

Reply:
IMO, in case of weakly typed structures it's perfectly okay to write
```go
// Cast is always successful since Put() only inserts []byte
value, _ = value.([]byte)
```
with no failure path, as it is not possible.